### DISCUSSION TITLE
Sorting Interfaces

### DISCUSSION THREAD

**Discussion by willflo** [netbox-community/netbox#7723] [Q&A]
First of all I would like to describe how we are using Netbox.

When documenting Switch stacks in Netbox we are currently creating a virtual chassis for them to see that they're logically one device.
We are naming our Switch Stacks in an order, where we name the single device Switch1-unit1 for the first device and Switch1-unit2 for the second device for example. The Interface are basically all the same, because we wanted to create standards for us. So there is not much room for individuality and errors.
All Interface on a Switch Stack are named like GigabitEthernet1/1/1 for example. It doesnt matter if they are on Unit2, they are always the same. 
When assigning a master device in the virtual chassis all the interfaces get included into the master device so you can see all of the interfaces under one device interface tab. The issue with this design is, that the sorting of the interfaces is kind of weird. 

![SortingInterfaces](https://user-images.githubusercontent.com/93641139/140022563-b35f0cb9-eccd-45ec-bd43-7b229f32425a.JPG)

I remember that in some older version the sorting was handled in the way that first the one node was displayed and after the second node was displayed.

When I try to sort via the device name the interfaces still get mixed up and are still not one after another.

Is there a possibility that I sort for A -> Device name and then B -> Interface name?

Thank you so much

**Comment by candlerb** [comment]
A switch *stack* is managed as a single entity - hence the name "virtual chassis".

If you do "show running-config" or equivalent on your switch stack's management plane, you won't see two interfaces called "GigabitEthernet1/1/1".  The switch will have some way of distinguishing them: for example, port 1 on the first switch might be "GigabitEthernet1/1/1", and port 1 on the second switch might be "GigabitEthernet2/1/1".  It's expected that you mirror this naming in Netbox.  

The easiest way to do it is to rename all the interfaces on the second unit *before* joining the two switches into a stack (using the bulk interface rename function).

Netbox to a degree does conflate the ideas of "physical interface" and "logical interface", because a single "Interface" object represents both.  Physically, port 1 may be labelled just "1" on the outside of the box, but logically it might be called "GigabitEthernet1/1/1" or "GigabitEthernet2/1/1" depending on which VC member it is in.  When you're telling someone to insert a cable in you'll say "plug into port 1 on Switch1-unit1"; but when you're configuring the switch you'll be configuring port GigabitEthernet1/1/1 or GigabitEthernet2/1/1.  The latter naming is what you need to use in Netbox.
